Install 3.1.1 on OSX with a homebrew installed Lua 5.3 fails to find Lua library

With Lua clean installed via Brew and luarocks installed clean (with no prior config):
```
➜  luarocks git:(master) ls $LUA_LIBDIR/liblua*            
lrwxr-xr-x  1 andrewstarks  admin  44 May 29 12:40 /usr/local/lib/liblua.5.3.5.dylib -> ../Cellar/lua/5.3.5_1/lib/liblua.5.3.5.dylib
lrwxr-xr-x  1 andrewstarks  admin  42 May 29 12:40 /usr/local/lib/liblua.5.3.dylib -> ../Cellar/lua/5.3.5_1/lib/liblua.5.3.dylib
lrwxr-xr-x  1 andrewstarks  admin  38 May 29 12:40 /usr/local/lib/liblua.dylib -> ../Cellar/lua/5.3.5_1/lib/liblua.dylib
lrwxr-xr-x  1 andrewstarks  admin  41 May 29 12:40 /usr/local/lib/liblua5.3.dylib -> ../Cellar/lua/5.3.5_1/lib/liblua5.3.dylib
➜  luarocks git:(master) ./configure                       

Configuring LuaRocks version dev...

Lua version detected: 5.3
Lua interpreter found: /usr/local/bin/lua5.3
lua.h found: /usr/local/include/lua5.3/lua.h
unzip found in PATH: /usr/bin

Done configuring.

LuaRocks will be installed at......: /usr/local
LuaRocks will install rocks at.....: /usr/local
LuaRocks configuration directory...: /usr/local/etc/luarocks
Using Lua from.....................: /usr/local

* Type make and make install:
  to install to /usr/local as usual.
* Type make bootstrap:
  to install LuaRocks into /usr/local as a rock.

➜  luarocks git:(master) make                  
mkdir -p "build"
rm -f src/luarocks/core/hardcoded.lua
echo "#!/bin/sh" > luarocks
echo "unset LUA_PATH LUA_PATH_5_2 LUA_PATH_5_3 LUA_PATH_5_4" >> luarocks
echo 'LUAROCKS_SYSCONFDIR="/usr/local/etc/luarocks" LUA_PATH="/Users/andrewstarks/src/luarocks/src/?.lua;;" exec "/usr/local/bin/lua5.3" "/Users/andrewstarks/src/luarocks/src/bin/luarocks" --project-tree="/Users/andrewstarks/src/luarocks/lua_modules" "$@"' >> luarocks
chmod +rx ./luarocks
./luarocks init

Initializing project 'luarocks' for Lua 5.3 ...
-----------------------------------------------

Checking your Lua installation ...

Error: Failed finding Lua library. You may need to configure LUA_LIBDIR.
make: *** [luarocks] Error 1
```
What am I missing?

--- a/src/luarocks/fs/unix.lua
+++ b/src/luarocks/fs/unix.lua
@@ -208,7 +208,7 @@ end
 
 function unix.is_dir(file)
    file = fs.absolute_name(file)
-   file = dir.normalize(file) .. "/"
+   file = dir.normalize(file) .. "/."
    local fd, _, code = io.open(file, "r")
    if code == 2 then -- "No such file or directory"
       return false

